doc/src/hexdoc_hexcasting/_templates/render.js.jinja

```diff
@@ -0,0 +1,371 @@
+
+  let load_func = (event) => {
+    const canvases = event.target.querySelectorAll("canvas");
+    canvases.forEach(initializeElem);
+  };
+
+  for (let collapsible of collapsibles) {
+    let patterns = Array.from(collapsible.getElementsByClassName("spell-viz"));
+
+    let open = collapsible.hasAttribute("open");
+
+    // @ts-ignore
+    collapsible.removeEventListener("toggle", last_load_func, {once: true});
+
+    if (!open) {
+      collapsible.addEventListener("toggle", load_func, {once:true})
+    }
+    
+    for (let pattern of patterns) {
+      let attr_to_transfer = ["data-start", "data-string", "width", "height", "data-per-world", "class"];
+      let canvas = document.createElement("canvas");
+      for (const index in attr_to_transfer) {
+        const attr = attr_to_transfer[index];
+        let value = pattern.getAttribute(attr);
+        canvas.setAttribute(attr, value ? value : "");
+      }
+
+      canvas.innerHTML = pattern.innerHTML;
+
+      collapsible.appendChild(canvas);
+
+      if (open) {
+        initializeElem(canvas);
+      }
+
+      pattern.remove()
+    }
+  }
+
+  last_load_func = load_func;
+
+}
+
+/** @type {Point} */
+const intersection_point = {
+  type: "Single",
+  marker: {
+    color: [255, 255, 255, 150],
+    radius: 0.07,
+  }
+};
+
+/** @type {Intersections} */
+const intersections = {
+  type: "EndsAndMiddle",
+  start: {
+    type: "BorderedMatch",
+    match_radius: 0.04,
+    border: intersection_point.marker
+  },
+  middle: intersection_point,
+  end: {
+    type: "Point",
+    point: intersection_point
+  }
+};
+
+const line_thickness = 0.12;
+
+/** @type {Color[]} */
+const colors = [
+  [214, 9, 177, 255],
+  [108, 25, 140, 255],
+  [50, 102, 207, 255],
+  [102, 110, 125, 255],
+];
+
+/** @type {Point} */
+const center_dot = {
+  type: "None"
+};
+
+/** @type {GridOptions} */
+const monocolor = {
+  line_thickness: line_thickness,
+  center_dot: center_dot,
+  pattern_options: {
+    type: "Uniform",
+    lines: {
+      type: "Monocolor",
+      color: colors[0],
+      bent: true
+    },
+    intersections: intersections,
+  },
+};
+
+/** @type {GridOptions} */
+const gradient = {
+  line_thickness: line_thickness,
+  center_dot: center_dot,
+  pattern_options: {
+    type: "Uniform",
+    lines: {
+      type: "Gradient",
+      colors: colors,
+      bent: true,
+      segments_per_color: 15,
+    },
+    intersections: intersections,
+  },
+};
+
+/** @type {GridOptions} */
+const segment = {
+  line_thickness: line_thickness,
+  pattern_options: {
+    type: "Uniform",
+    intersections: intersections,
+    lines: {
+      type: "SegmentColors",
+      colors: colors,
+      triangles: {
+        type: "BorderStartMatch",
+        match_radius: 0.13,
+        border: {
+          color: [255, 255, 255, 255],
+          radius: 0.20,
+        }
+      },
+      collisions: {
+        type: "Dashes",
+        color: [255, 0, 0, 255],
```

Comment:
I'm fine with either. I set it to dashes because I felt like it some might see it and think they can draw it somehow, but I'm not sure how much of an issue that is.

Reply:
mmm.. I think the onus to document that is more on whoever's putting impossible patterns in their book. I vote for the parallel lines, they look neat :)

Reply:
Should it be the standard Parallel Lines option? The one that doesn't add the dashes when there are too many? Or should it still include a dashes color (as a backup)? Alternatively, it could use matched dashes that shows a bar for each color.